### PR TITLE
Awards page

### PR DESCRIPTION
**Pull request by lucasreljic**
- Updated old Awards branch
- Cleaned up and organized working directory
- Refactored the Awards page

- - -
This change is [Reviewable](https://reviewable.io/reviews/waterloo-rocketry/website-react/21)

### CODE REVIEW

**Review by KavinSatheeskumar — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/website-react/21)* status: 0 of 11 files reviewed, 3 unresolved discussions (waiting on @lucasreljic)

___
*[`package-lock.json` line 5 at r2](https://reviewable.io/reviews/waterloo-rocketry/website-react/21#-NMPo60r9K_1Fsv0yqoX:-NMPo60r9K_1Fsv0yqoY:bbxg8jr) ([raw file](https://github.com/waterloo-rocketry/website-react/blob/853b340d9a0123a9e65839416c383c7a35c78d45/package-lock.json#L5)):*
> ```JSON
>   "version": "0.1.0",
>   "lockfileVersion": 2,
>   "requires": true,
> ```

What happened here?
___
*[`.vscode/settings.json` line 9 at r2](https://reviewable.io/reviews/waterloo-rocketry/website-react/21#-NMPoPg8-lFFbkZKTe6c:-NMPoPg8-lFFbkZKTe6d:b-k2d9h5) ([raw file](https://github.com/waterloo-rocketry/website-react/blob/204eabf04b3d82f375fa10aac95e2a940ee7fe70/.vscode/settings.json#L9)):*
> ```JSON
>         100
>     ],
>     "git.ignoreLimitWarning": true
> ```

I think the .vscode folder should be git ignored as a whole
___
*[`src/index.jsx` line 63 at r2](https://reviewable.io/reviews/waterloo-rocketry/website-react/21#-NMPoiP6-NEct1uo-ORP:-NMPoiP6-NEct1uo-ORQ:b-ih7rnf) ([raw file](https://github.com/waterloo-rocketry/website-react/blob/204eabf04b3d82f375fa10aac95e2a940ee7fe70/src/index.jsx#L63)):*
> ```JavaScript
>           element={<Page title="Sponsors"><Sponsors /></Page>}
>         />
>       </Routes>
> ```

Yeah ok, so I fixed the formatting of this component in another PR that hasn't been merged. Just letting you know that this entire file is wrong. 

I'll fix it when I merge that original PR. so for now, dw about this

**Review by QuantumManiac — CHANGES_REQUESTED** [review]
> Reviewed 6 of 11 files at r1, 5 of 5 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/website-react/21)* status: all files reviewed, 6 unresolved discussions (waiting on @KavinSatheeskumar and @lucasreljic)

___
*[`package-lock.json` line 5 at r2](https://reviewable.io/reviews/waterloo-rocketry/website-react/21#-NMPo60r9K_1Fsv0yqoX:-NMQA8rJ-9h62GChMy_3:b-fnrevk) ([raw file](https://github.com/waterloo-rocketry/website-react/blob/853b340d9a0123a9e65839416c383c7a35c78d45/package-lock.json#L5)):*
<details><summary><i>Previously, KavinSatheeskumar (Kavin Satheeskumar) wrote…</i></summary><blockquote>

What happened here?
</blockquote></details>

I don't seem to be having any issues because of it so let's assume that it's fine
___
*[`src/components/AwardItem/AwardItem.jsx` line 20 at r2](https://reviewable.io/reviews/waterloo-rocketry/website-react/21#-NMQB76V7Esx5Saq2EL9:-NMQB76V7Esx5Saq2ELA:bpkql7v) ([raw file](https://github.com/waterloo-rocketry/website-react/blob/204eabf04b3d82f375fa10aac95e2a940ee7fe70/src/components/AwardItem/AwardItem.jsx#L20)):*
> ```JavaScript
> };
> 
> // Component to add a singular award
> ```

Very good idea to actually document our components and the parameters it requires here. That being said, it might be a better idea to document them using something a bit more standardized and commonplace, such as JSDoc. I've not actually used it to document React components so a bit of research is probably in order. A quick google search brings this page up: https://www.inkoop.io/blog/a-guide-to-js-docs-for-react-js/
___
*[`src/components/AwardItem/AwardItem.jsx` line 39 at r2](https://reviewable.io/reviews/waterloo-rocketry/website-react/21#-NMQCh7l3oAaB-TCMkkZ:-NMQCh7m5-b51Nt8JWOO:bl9r540) ([raw file](https://github.com/waterloo-rocketry/website-react/blob/204eabf04b3d82f375fa10aac95e2a940ee7fe70/src/components/AwardItem/AwardItem.jsx#L39)):*
> ```JavaScript
>         <div className={styles.content_text}>
>           <p>
>             {AwardDescription}
> ```

I feel that instead of passing the Award description as a prop you could instead throw it in as a child. See any other component that uses the `children` prop for examples
___
*[`src/components/AwardItem/AwardItem.jsx` line 49 at r2](https://reviewable.io/reviews/waterloo-rocketry/website-react/21#-NMQCGavDh4xuGdqYR4E:-NMQCGavDh4xuGdqYR4F:b-psdwtk) ([raw file](https://github.com/waterloo-rocketry/website-react/blob/204eabf04b3d82f375fa10aac95e2a940ee7fe70/src/components/AwardItem/AwardItem.jsx#L49)):*
> ```JavaScript
>   );
> };
> AwardItem.defaultProps = {
> ```

Hmm, we don't actually specify `defaultProps` in any other components. Maybe we should be?

File: src/components/AwardItem/AwardItem.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    </div>
+  );
+};
+AwardYear.defaultProps = {
+  Year: 9999,
+};
+
+/**
+* To add a singular award
+* @param  {string} RocketName Name of Rocket (Str)
```
Comment (QuantumManiac) [inline src/components/AwardItem/AwardItem.jsx:24]:
```suggestion
* @param  {string} RocketName Name of Rocket
```
Can probably remove the old way you specified the type considering that the JSDoc syntax does it itself.

**Review by QuantumManiac — APPROVED** [review]
Wow, I love how reviewable still doesn't integrate with GitHub's code suggestions feature. 

Just that small fix to your comments. Otherwise, LGTM!

**Review by KavinSatheeskumar — APPROVED** [review]
:lgtm:

> *[Reviewable](https://reviewable.io/reviews/waterloo-rocketry/website-react/21#-:-NMo-TeO0-mNuw0jANow:bnfp4nl)* status: 8 of 12 files reviewed, 7 unresolved discussions (waiting on @lucasreljic and @QuantumManiac)

**Pull request merged**